models/cice/README.rst

```diff
@@ -0,0 +1,201 @@
+
+Contents
+========
+
+#. `Overview`_
+#. `Development Notes`_
+#. `Namelist`_
+#. `Terms of Use`_
+#. `References`_
+
+Overview
+========
+
+The `Community Ice CodE (CICE) <https://github.com/CICE-Consortium/CICE>`_ is a
+sea ice model that was first developed by Elizabeth Hunke as the Los Alamos Sea
+Ice Model. Its code base and capabilities have grown as a result of continued
+development by the broader geosciences community, an effort organized by the
+CICE Consortium.
+
+The DART model interface was developed to work with CICE's dynamical core on an
+Arakawa B-grid. [1]_ When CICE is coupled to POP in CESM, the ocean and sea ice
+grids are identical.
+
+According to the CICE manual:
+
+  The spatial discretization is specialized for a generalized orthogonal B-grid
+  as in Murray (1996) [2]_ or Smith et al. (1995). [3]_ The ice and snow area,
+  volume and energy are given at the center of the cell, velocity is defined at
+  the corners, and the internal ice stress tensor takes four different values
+  within a grid cell; bilinear approximations are used for the stress tensor
+  and the ice velocity across the cell, as described in Hunke and Dukowicz
+  (2002). [4]_ This tends to avoid the grid decoupling problems associated with
+  the B-grid.
+
+Hence, in the DART interface:
+
+- U, V are at grid cell corners
+- T, h, hs, and the various scalar quantities are at grid cell centers
+
+Development Notes
+=================
+
+CICE is under active development to work with other grids, such as the
+unstructured grid in MPAS and the C-grid in MOM. Due to this activity, this
+README contains development notes chronicling the development of the model
+interface. We ask that developers continue to document the development both in
+this README and with descriptive comments in the source code.
+
+- Possible bug found in ``model_mod.f90`` for POP where set_date is sent sec
+  this day. The routine wants sec this min.
+
+Notes from Cecilia M. Bitz on 14 May 2016
+-----------------------------------------
+
+- Created ``../../obs_def/obs_def_cice_mod.f90`` to make new obs_kinds used in
+  ``model_mod.f90`` and ``input.nml``.
+- Not sure about ``QTY_TRACERARRAY_CATS``
+- Model mod assumes that the grid is identical to POP
+- Leaving this part but it may be unneeded in CICE ``INTERFACE
+  vector_to_prog_var MODULE PROCEDURE vector_to_2d_prog_var
+  MODULE PROCEDURE vector_to_3d_prog_var END INTERFACE``
+- Not used in pop so of course not used now in CICE either, why? ``subroutine
+  vector_to_3d_prog_var(x, varindex,
+  data_3d_array) subroutine get_gridsize(num_x, num_y, num_z)``
+- Come back here, some changes made below but need to look line-by-line still
+  ``subroutine get_state_meta_data(state_handle, index_in, location,
+  var_type)``
+
+Fortran Files
+~~~~~~~~~~~~~
+
+- ``dart_to_cice.f90`` Think it is done
+- ``cice_to_dart.f90`` Is trivial so hope it's done too
+- ``model_mod_check.f90``
+- ``dart_cice_mod.f90`` Should it have a get_cat_dim?
```

Comment:
We need to test this for Manhattan-readiness and delete the extraneous routines if they are not needed.  I suspect the 'cice_to_dart.f90' is not needed, but the 'dart_to_cice.f90' might have a partitioning function.  Who do we know that is assimilating with this model?

Reply:
The content in the CICE documentation is several years old. I think we should update it in a separate pull request.